core/server/web/api/canary/admin/routes.js

```diff
@@ -17,6 +17,46 @@ module.exports = function apiRoutes() {
 
     const http = apiCanary.http;
 
+    (function ghostAuthModule(router) {
```

Comment:
Obviously this block has to move somewhere else, but I'm not quite sure where. Would it make sense to this to live as one of the "controllers" with a query method and access to `frame` - or in the "services" dir like the session stuff :thinking: 

@gargol Would be interested to hear your thoughts

Reply:
@allouis is this endpoint meant to become a part of our Admin API or is there going to be a new concept of "Aauth" endpoints for this one and the `/.well-known` one? This code can potentially have a controller, but don't know if it necessarily fits with the rest of them for it's special purpose :thinking:

Reply:
Hmm, I guess so? It's gonna be used by the admin client to fetch tokens. And we're gonna want to authenticate based on the user logged into the admin - so yeah I think it will be.

Yeah doesn't really seem like our controller concept works for this kind of thing - though I guess we could do it like the session controller is done, as a passthrough?

Reply:
I don't see why not making it a passthough? The session example is a good one in this case. Reduces all the unneeded fluff around validations and such.. Is this token generation limited to specific user roles only (then we might need to have permission checks)?

Reply:
Yeah I think we're gonna want to either:
  1. Only issue to "Owner" role.
  2. Include the role of the user in the token.

Do you know how easy it is to get the role of a user to achieve 2?

Reply:
@gargol @allouis Think this is the only piece left to finalize before we can merge the PRs which will put billing logic behind the config flags. I am not sure if this is something that can still exist here for now and merged, and put in the right place when we revisit it, or if its something that needs to happen before merge. Will be interested in your thoughts on how to move forward with this :)

Reply:
@rishabhgrg don't think we want to rush something like this into the core. It has no tests and is not properly structure. It needs to be extracted into a separate controller and proper permission check should be in place with minimal tests. 

Was thinking it could take similar form of the member signing url resource structure - https://github.com/TryGhost/Ghost/blob/master/core/server/api/canary/memberSigninUrls.js

Reply:
👍  Yeah agreed, atleast as long as is its directly exposed its better to be in right structure. Think the signing url structure is similar to the sessions approach as well which was discussed above, will see if this is something we can pick up in cleanup.

Reply:
I'm already context loaded on this one, so was planning to tackle it tomorrow ;)

Reply:
@gargol Ah sure, sounds good 😄

Reply:
> Include the role of the user in the token.

@allouis do we really need to include the role right now if we restrict access to the endpoint to "owneres" only?

Reply:
It was an either/or situation :relaxed: https://github.com/TryGhost/Ghost/pull/11559#discussion_r386947240 your proposal here should give us everything we need :)